presto-expressions/src/test/java/com/facebook/presto/expressions/translator/TestRowExpressionTranslator.java

```diff
@@ -0,0 +1,224 @@
+            this.functionManager = functionManager;
+        }
+
+        @Override
+        public Optional<String> translateConstant(ConstantExpression literal, Map<VariableReferenceExpression, ColumnHandle> context)
+        {
+           return Optional.of(literal.toString());
+        }
+
+        @Override
+        public Optional<String> translateCall(FunctionHandle functionHandle, List<TranslatedExpression<String>> translatedExpressions, Map<VariableReferenceExpression, ColumnHandle> context)
+        {
+            FunctionMetadata functionMetadata = functionManager.getFunctionMetadata(functionHandle);
+            return functionTranslator.translate(functionMetadata, translatedExpressions);
+        }
+
+        @Override
+        public Optional<String> translateSpecialForm(SpecialFormExpression.Form form, List<TranslatedExpression<String>> translatedExpressions, Map<VariableReferenceExpression, ColumnHandle> context)
+        {
+            if (!form.equals(SpecialFormExpression.Form.AND)) {
+                return Optional.empty();
+            }
+            return Optional.of(translatedExpressions.get(0).getTranslated().get() + " TEST_AND " + translatedExpressions.get(1).getTranslated().get());
+        }
+
+        @Override
+        public Optional<String> translateVariable(VariableReferenceExpression reference, Map<VariableReferenceExpression, ColumnHandle> context)
+        {
+            return Optional.of(reference.getName());
+        }
+    }
+
+    private static class TestFunctions {
+        @ScalarFunction
+        @SqlType(StandardTypes.BIGINT)
+        public static String bitwiseAnd(@SqlType(StandardTypes.BIGINT) TranslatedExpression<String> sql1, @SqlType(StandardTypes.BIGINT) TranslatedExpression<String> sql2)
+        {
+            return sql1.getTranslated().get() + " BITWISE_AND " + sql2.getTranslated().get();
+        }
+
+        @ScalarFunction("ln")
+        @SqlType(StandardTypes.DOUBLE)
+        public static String ln(@SqlType(StandardTypes.DOUBLE) TranslatedExpression<String> sql)
+        {
+            return "LNof(" + sql.getTranslated().get() + ")";
+        }
+
+    }
+
+    @Test
+    public void testEndToEndFunctionTranslation() {
+        String untranslated = "bitwise_and(1, 2)";
+        CallExpression callExpression = (CallExpression) sqlToRowExpressionTranslator.translate(expression(untranslated), TypeProvider.empty());
+
+        TranslatedExpression translatedExpression = translateWith(
+                callExpression,
+                determinismEvaluator,
+                functionResolution,
+                functionManager,
+                new TestFunctionTranslator(functionManager, buildFunctionTranslator(ImmutableSet.of(TestFunctions.class))),
+                emptyMap());
+        assertEquals(translatedExpression.getTranslated().get(), "1 BITWISE_AND 2");
+    }
+
+    @Test
+    public void testEndToEndFunctionTranslation2() {
```

Comment:
Combine `testEndToEndFunctionTranslation` and `testEndToEndFunctionTranslation2` into one